include/dsn/tool-api/rpc_message.h

```diff
@@ -175,6 +175,11 @@ class message_ex : public ref_counter,
                                                        int size,
                                                        int thread_hash = 0,
                                                        uint64_t partition_hash = 0);
+    // Used for bulk load ingestion
+    // meta send bulk load ingestion request to primary, primary will consider it as a special write
+    // request, in other words, create a request which will be considered as sent from user client
+    DSN_API static message_ex *
+    create_client_request(dsn::task_code rpc_code, dsn::gpid pid, int timeout_milliseconds = 0);
```

Comment:
If this is for bulk-load only, put it on replica/bulkload folder. Don't add upper-level details in a low-level module.

Reply:
BTW. This function seems touching no private member. Why not make it a global&static function?

Reply:
I remove this function.